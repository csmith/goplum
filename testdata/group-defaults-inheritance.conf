alert debug.sysout "debug" {}

group "test-group" {
  alert_limit = 5
  alert_window = 10m
  
  defaults {
    interval = 45s
    timeout = 15s
    alerts = ["debug"]
  }
}

check debug.random "test" {
  groups = ["test-group"]
}