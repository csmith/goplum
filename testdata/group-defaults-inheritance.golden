{
  "Alerts": {
    "debug": {}
  },
  "Checks": {
    "test": {
      "Name": "test",
      "Type": "debug.random",
      "Config": {
        "Alerts": [
          "debug"
        ],
        "Groups": [
          "test-group"
        ],
        "Interval": 45000000000,
        "Timeout": 15000000000,
        "GoodThreshold": 2,
        "FailingThreshold": 2
      },
      "Check": {
        "PercentGood": 0.5
      },
      "LastRun": "0001-01-01T00:00:00Z",
      "Scheduled": false,
      "Settled": false,
      "State": "indeterminate",
      "Suspended": false,
      "History": [
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null,
        null
      ]
    }
  },
  "Groups": {
    "test-group": {
      "Name": "test-group",
      "AlertLimit": 5,
      "AlertWindow": 600000000000,
      "Defaults": {
        "Alerts": [
          "debug"
        ],
        "Groups": null,
        "Interval": 45000000000,
        "Timeout": 15000000000,
        "GoodThreshold": 0,
        "FailingThreshold": 0
      }
    }
  }
}